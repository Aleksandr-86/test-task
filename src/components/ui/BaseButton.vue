<script setup lang="ts">
defineProps<{
  label: string
  iconSrc?: string
}>()
</script>

<template>
  <button
    class="flex h-10 w-full cursor-pointer items-center justify-center rounded-md bg-blue-700 text-2xl font-bold text-gray-200 dark:bg-blue-100 dark:text-gray-700"
    type="button">
    <span v-if="iconSrc" class="mr-2 h-8 w-8">
      <img :src="iconSrc" />
    </span>
    {{ label }}
  </button>
</template>
