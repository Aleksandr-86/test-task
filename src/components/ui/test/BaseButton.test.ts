import { render } from '@testing-library/vue'
import BaseButton from '@/components/ui/BaseButton.vue'

describe('Компонент BaseButton', () => {
  it('структура соответствует снимку', () => {
    const wrapper = render(BaseButton, {
      props: {
        modelValue: true,
        value: true,
        label: 'mockLabel'
      }
    })

    expect(wrapper).toMatchSnapshot()
  })
})
