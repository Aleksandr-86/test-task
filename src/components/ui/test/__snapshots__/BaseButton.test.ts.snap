// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Компонент BaseButton > структура соответствует снимку 1`] = `
{
  "baseElement": <body>
    <div>
      <button
        class="flex h-12 w-full cursor-pointer items-center justify-center rounded-lg bg-blue-700 text-2xl font-bold text-gray-200 dark:bg-blue-100 dark:text-gray-700"
        modelvalue="true"
        type="button"
        value="true"
      >
        <!--v-if-->
         mockLabel
      </button>
    </div>
  </body>,
  "container": <div>
    <button
      class="flex h-12 w-full cursor-pointer items-center justify-center rounded-lg bg-blue-700 text-2xl font-bold text-gray-200 dark:bg-blue-100 dark:text-gray-700"
      modelvalue="true"
      type="button"
      value="true"
    >
      <!--v-if-->
       mockLabel
    </button>
  </div>,
  "debug": [Function],
  "emitted": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "html": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
